rental.rb

```diff
@@ -0,0 +1,21 @@
+require_relative 'book'
+
+class Rental
+  attr_accessor :date, :books, :person
+
+  def initialize(date, books, person)
+    @date = date
+    @books = books
+    @person = person
+  end
+
+  def book_rental=(book)
+    @books = book
+    book.book_rentals.push(self) unless book.book_rentals.include?(self)
+  end
+
+  def person_rental=(person)
+    @person = person
+    person.person_rentals.push(self) unless person.person_rentals.include?(self)
+  end
```

Comment:
- Hey, while you've added some methods which is great, however, this is more not consistent as your Rental should only take responsibility for pushing into to instance variable but in your case, you've added extra methods for that. you can further simplify it to:

```ruby
class Rental
  attr_accessor :date, :person, :book

  def initialize(date, person, book)
    @date = date
    @person = person
    @book = book
    person.rentals << self
    book.rentals << self
  end
end
```

And this does all the heavy lifting for you. please, let's improve on this. whcn you make this changes, be sure to update other part of your codes.

Reply:
Done!